nucleus/common/simple-glassfish-api/src/main/java/org/glassfish/embeddable/GlassFishProperties.java

```diff
@@ -74,8 +74,12 @@ public GlassFishProperties() {
      * for the plugged in {@link GlassFishRuntime} (if any)
      *
      * @param props Properties object which will back this GlassFishProperties object.
+     * @throws IllegalArgumentException if props is null
      */
     public GlassFishProperties(Properties props) {
+        if(props == null){
```

Comment:
this should be `Objects.requireNonNull(props, "props should not be null")`

Reply:
or something similar to that, introducing a different exception type might break some functionality and introduces unnecessary risks